cmd/tsbs_generate_queries/databases/timescaledb/devops_test.go

```diff
@@ -152,3 +154,333 @@ func TestDevopsGetSelectClausesAggMetrics(t *testing.T) {
+				"WHERE c.tags_id = t.id ORDER BY time DESC LIMIT 1) AS b ON true ORDER BY t.hostname, b.time DESC",
+		},
+	}
+
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(13 * time.Hour)
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			d := NewDevops(s, e, 10)
+			d.UseJSON = c.useJSON
+			d.UseTags = c.useTags
+
+			q := d.GenerateEmptyQuery()
+			d.LastPointPerHost(q)
+			verifyQuery(t, q, c.expectedHumanLabel, c.expectedHumanDesc, c.expectedHypertable, c.expectedSQLQuery)
+		})
+	}
+}
+
+func TestHighCPUForHosts(t *testing.T) {
+	cases := []struct {
+		desc               string
+		nHosts             int
+		expectedHumanLabel string
+		expectedHumanDesc  string
+		expectedHypertable string
+		expectedSQLQuery   string
+	}{
+		{
+			desc:               "zero hosts",
+			nHosts:             0,
+			expectedHumanLabel: "TimescaleDB CPU over threshold, all hosts",
+			expectedHumanDesc:  "TimescaleDB CPU over threshold, all hosts: 1970-01-01T00:16:22Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT * FROM cpu WHERE usage_user > 90.0 and time >= '1970-01-01 00:16:22.646325 +0000'" +
+				" AND time < '1970-01-01 12:16:22.646325 +0000' ",
+		},
+		{
+			desc:               "one host",
+			nHosts:             1,
+			expectedHumanLabel: "TimescaleDB CPU over threshold, 1 host(s)",
+			expectedHumanDesc:  "TimescaleDB CPU over threshold, 1 host(s): 1970-01-01T00:47:30Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT * FROM cpu WHERE usage_user > 90.0 and time >= '1970-01-01 00:47:30.894865 +0000'" +
+				" AND time < '1970-01-01 12:47:30.894865 +0000' AND (hostname = 'host_9')",
+		},
+		{
+			desc:               "five hosts",
+			nHosts:             5,
+			expectedHumanLabel: "TimescaleDB CPU over threshold, 5 host(s)",
+			expectedHumanDesc:  "TimescaleDB CPU over threshold, 5 host(s): 1970-01-01T00:08:59Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT * FROM cpu WHERE usage_user > 90.0 and time >= '1970-01-01 00:08:59.080812 +0000'" +
+				" AND time < '1970-01-01 12:08:59.080812 +0000' AND (hostname = 'host_5' OR hostname = 'host_9' " +
+				"OR hostname = 'host_1' OR hostname = 'host_7' OR hostname = 'host_2')",
+		},
+	}
+
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(13 * time.Hour)
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			d := NewDevops(s, e, 10)
+
+			q := d.GenerateEmptyQuery()
+			d.HighCPUForHosts(q, c.nHosts)
+
+			verifyQuery(t, q, c.expectedHumanLabel, c.expectedHumanDesc, c.expectedHypertable, c.expectedSQLQuery)
+		})
+	}
+}
+
+func verifyQuery(t *testing.T, q query.Query, humanLabel, humanDesc, hypertable, sqlQuery string) {
+	tsq, ok := q.(*query.TimescaleDB)
+
+	if !ok {
+		t.Error("Filled query is not *query.TimescaleDB type")
+		return
+	}
+
+	if string(tsq.HumanLabel) != humanLabel {
```

Comment:
personally a fan of the form:
```
if got := string(tsq.HumanLabel); got != humanLabel {
  t.Errorf("incorrect HumanLabel: got\n%s\nwant\n%s", got, humanLabel)
}
```

Maybe you can break the printing into a few lines, but I usually just do one. I would prefer to use the `got`/`want` naming instead of `got`/`expected` since we use `want` everywhere else

Reply:
Updated to recommendation.